pretyping/typeclasses.mli

```diff
@@ -123,5 +123,5 @@ val resolve_typeclasses : ?filter:evar_filter -> ?unique:bool ->
   ?split:bool -> ?fail:bool -> env -> evar_map -> evar_map
 val resolve_one_typeclass : ?unique:bool -> env -> evar_map -> EConstr.types -> evar_map * EConstr.constr
 
-val solve_all_instances_hook : (env -> evar_map -> evar_filter -> bool -> bool -> bool -> evar_map) Hook.t
-val solve_one_instance_hook : (env -> evar_map -> EConstr.types -> bool -> evar_map * EConstr.constr) Hook.t
+val set_solve_all_instances : (env -> evar_map -> evar_filter -> bool -> bool -> bool -> evar_map) -> unit
```

Comment:
As these refs are not handled by the summary (neither by the proof engine) their intended use should be documented.

For example, if they can be set by a plugin indeed they should be able to be undone plugin unloading (then they should be summary-tracked I guess)

Reply:
@SkySkimmer had a proposal for that, as of today we don't have the machinery

Reply:
Note that I'm requesting to provide a spec in form of documentation, then we can see what machinery we need.

Reply:
ok then